misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/codeaction/CodeActionTest.java

```diff
@@ -0,0 +1,259 @@
+ * herein is strictly forbidden, unless permitted by WSO2 in accordance with
+ * the WSO2 Commercial License available at http://wso2.com/licenses.
+ * For specific language governing the permissions and limitations under
+ * this license, please see the license as well as any agreement you’ve
+ * entered into with WSO2 governing the purchase of this software and any
+ */
+package org.ballerinalang.datamapper.codeaction;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.ballerinalang.datamapper.util.FileUtils;
+import org.ballerinalang.datamapper.util.TestUtil;
+import org.ballerinalang.datamapper.utils.HttpClientRequest;
+import org.ballerinalang.datamapper.utils.HttpResponse;
+import org.ballerinalang.langserver.codeaction.CodeActionUtil;
+import org.eclipse.lsp4j.CodeActionContext;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.Endpoint;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Matchers.any;
+
+/**
+ * Test Cases for CodeActions.
+ *
+ * @since 2.0.0
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(HttpClientRequest.class)
+public class CodeActionTest extends PowerMockTestCase {
+
+    private Endpoint serviceEndpoint;
+
+    private JsonParser parser = new JsonParser();
+
+    private Path sourcesPath = new File(getClass().getClassLoader().getResource("codeaction").getFile()).toPath();
+
+    private static final Logger log = LoggerFactory.getLogger(CodeActionTest.class);
+    private static final int HTTP_200_OK = 200;
+    private static final int HTTP_422_UN_PROCESSABLE_ENTITY = 422;
+    private static final int HTTP_500_INTERNAL_SERVER_ERROR = 500;
+
+    @BeforeClass
+    private void init() {
+        this.serviceEndpoint = TestUtil.initializeLanguageSever();
+        String startConfigPath = "codeaction" + File.separator + "config" + File.separator + "startConfig.json";
+        JsonObject configs = FileUtils.fileContentAsObject(startConfigPath);
+        TestUtil.setWorkspaceConfig(serviceEndpoint, configs);
+    }
+
+    @Test(dataProvider = "codeAction-data-mapper-data-provider")
+    public void testDataMapperCodeAction(String config, String source) throws Exception {
```

Comment:
this test method is bit long, commenting would helpful for  readability

Reply:
Done